### PR TITLE
Modal tests

### PR DESCRIPTION
**Pull request by Innokentiy945**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,9 +4,16 @@
 * Support [Angular materials](https://material.angular.io/components/categories) elements
 * All tasks [angular](https://github.com/jdi-testing/jdi-light/issues?q=is%3Aissue+label%3AAngular+)
 
+##1.3.4 (06.02.2021)
```
Comment (alexarm) [inline CHANGELOG.md:7]: Are this changes required?

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import org.testng.annotations.Test;
+
+import static io.github.com.StaticSite.modalFrame;
+import static io.github.com.StaticSite.simpleModalPage;
+import static io.github.com.StaticSite.springModalPage;
+import static io.github.com.StaticSite.transitionsModalPage;
+
+public class ModalTests extends TestsInit {
+
+    private String expectedSimpleDescription = "Duis mollis, est non commodo luctus, nisi erat porttitor ligula.";
```
Comment (alexarm) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java:13]: Better use private static final

File: jdi-light/src/main/java/com/epam/jdi/light/elements/base/JdiSettings.java [hunk 3]
```diff
@@ -28,6 +28,7 @@
 import static com.epam.jdi.tools.ReflectionUtils.isInterface;
 import static java.lang.String.format;
 import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
```
Comment (alexarm) [inline jdi-light/src/main/java/com/epam/jdi/light/elements/base/JdiSettings.java:31]: Is this required?

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @JDIAction("Is '{name}' opened")
+    public boolean isTransitionModalOpened() {
+        return core().hasClass("makeStyles-paper-108");
+    }
+
+    @JDIAction("Get '{name}' description")
+    public boolean getTransitionModalDescription(String description) {
+        return core().find(By.id("transition-modal-description")).getText().equals(description);
```
Comment (alexarm) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:37]: Is this Id used in every modal in MaterualUI?

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+import com.epam.jdi.light.common.JDIAction;
+import com.epam.jdi.light.elements.base.UIBaseElement;
+import com.epam.jdi.light.material.asserts.utils.ModalAssert;
+import org.openqa.selenium.By;
+
+public class Modal extends UIBaseElement<ModalAssert> {
+
+    @JDIAction("Click on '{name}'")
+    public void clickOn() {
```
Comment (alexarm) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:11]: Better implement HasClick interface from JDI which already has this method

**Review by alexarm — CHANGES_REQUESTED** [review]

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java (hunk 4 above)
Comment (Innokentiy945) [reply jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:37]: Yes, look this:
https://material-ui.com/components/modal/

File: jdi-light/src/main/java/com/epam/jdi/light/elements/base/JdiSettings.java (hunk 3 above)
Comment (Innokentiy945) [reply jdi-light/src/main/java/com/epam/jdi/light/elements/base/JdiSettings.java:31]: rolled back

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java (hunk 2 above)
Comment (Innokentiy945) [reply jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java:13]: Fixed

File: CHANGELOG.md (hunk 1 above)
Comment (Innokentiy945) [reply CHANGELOG.md:7]: rolled back

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java (hunk 5 above)
Comment (Innokentiy945) [reply jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:11]: Implemented

File: jdi-light-material-ui-tests/src/main/java/io/github/com/StaticSite.java [hunk 6]
```diff
@@ -281,6 +294,10 @@
     public static WebPage utilsPopoverHoverPage;
 
     // login page
+
+    @Url("/story/material-ui-utils-modal--server-modal")
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/main/java/io/github/com/StaticSite.java:298]: Please rewrite like Url("/material-ui-utils-modal--server-modal")

File: jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package io.github.com.pages.utils;
+
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.Css;
+import com.epam.jdi.light.material.elements.utils.Modal;
+
+public class ModalFrame {
+
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java:7]: Please add button element for opening Modal

File: jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+    @Css("#root > div > button")
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java:8]: Please use JDIModal annotation

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    private static final String expectedSimpleTitle = "Text in a modal";
+    private static final String expectedSimpleDescription = "Duis mollis, est non commodo luctus, nisi erat porttitor ligula.";
+    private static final String expectedTransitionTitle = "Transition modal";
+    private static final String expectedTransitionDescription = "react-transition-group animates me.";
+    private static final String expectedServerTitle = "Server-side modal";
+    private static final String expectedServerDescription = "If you disable JavaScript, you will still see me.";
+
+
+    @Test
+    public void simpleModalTest() {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java:22]: Please fix tests. It doesn't work now

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        transitionsModalPage.shouldBeOpened();
+
+        modalFrame.modal.getModalButton();
+        modalFrame.modal.is().verifyingTitle(expectedTransitionTitle);
+        modalFrame.modal.is().verifyingDescription(expectedTransitionDescription);
+    }
+
+    @Test
+    public void serverSideModalTests() {
+        serverSideModalPage.shouldBeOpened();
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java:43]: Please use open() instead of shouldBeOpened()

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+        modalFrame.modal.getModalButton();
+        modalFrame.modal.is().verifyingTitle(expectedSimpleTitle);
+        modalFrame.modal.is().verifyingDescription(expectedSimpleDescription);
+        modalFrame.modal.getInnerModalWindow();
+        modalFrame.modal.is().verifyInnerModalPageOpened();
+    }
+
+    @Test
+    public void transitionModalTests() {
+        transitionsModalPage.shouldBeOpened();
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java:34]: Please use open() instead of shouldBeOpened()

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    private static final String expectedSimpleDescription = "Duis mollis, est non commodo luctus, nisi erat porttitor ligula.";
+    private static final String expectedTransitionTitle = "Transition modal";
+    private static final String expectedTransitionDescription = "react-transition-group animates me.";
+    private static final String expectedServerTitle = "Server-side modal";
+    private static final String expectedServerDescription = "If you disable JavaScript, you will still see me.";
+
+
+    @Test
+    public void simpleModalTest() {
+        simpleModalPage.shouldBeOpened();
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java:23]: Please use open() instead of shouldBeOpened()

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/annotations/JDIModal.java [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.FIELD})
+public @interface JDIModal {
+
+    @MarkupLocator String modalTitle() default "simple-modal-title";
+    @MarkupLocator String modalDescription() default "simple-modal-description";
+    @MarkupLocator String modalInnerButton() default "//button[contains('type=\"button\"')]";
+    @MarkupLocator String innerModalPage() default  "body > div:nth-child(10) > div.makeStyles-paper-32";
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/annotations/JDIModal.java:17]: Please delete innerModalPage() field and add root(). Also, set all fields empty by default

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+import com.epam.jdi.light.material.elements.utils.Modal;
+import org.hamcrest.Matchers;
+
+import static com.epam.jdi.light.asserts.core.SoftAssert.jdiAssert;
+
+public class ModalAssert extends UIAssert<ModalAssert, Modal> {
+
+    private String expectedTitle;
+    private String expectedDescription;
+
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java:14]: Please remove:     
    private String expectedTitle;
    private String expectedDescription;

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingTitle(String actualTitle) {
+        jdiAssert(element().verifyModalTitle(actualTitle), Matchers.is(expectedTitle));
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java:17]: Rewrite it:
jdiAssert(element().verifyModalTitle(actualTitle), Matchers.is(true));

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+        return this;
+    }
+
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingDescription(String actualDescription) {
+        jdiAssert(element().verifyModalDescription(actualDescription), Matchers.is(expectedDescription));
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java:23]: Rewrite it:
        jdiAssert(element().verifyModalDescription(actualDescription), Matchers.is(true));

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+        return this;
+    }
+
+    @JDIAction("Assert that '{name} opened")
+    public ModalAssert verifyInnerModalPageOpened() {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java:28]: Please remove this method

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+    @JDIAction("Get '{name}'")
+    public boolean verifyModalDescription(String actualDescription) {
+        return core().find(modalDescription).getText().equals(actualDescription);
+    }
+
+    @JDIAction("Get '{name}'")
+    public void getInnerModalWindow() {
+        core().find(modalInnerButton);
+    }
+
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:49]: Please add:

  @Override
  public ModalAssert is() {
    return new ModalAssert().set(this);
  }

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+public class Modal extends UIBaseElement<ModalAssert> implements HasClick, ISetup {
+
+    String modalTitle;
+    String modalDescription;
+    String modalInnerButton;
+    String innerModalPage;
+
+    @JDIAction("Is '{name}' enabled")
+    public void getModalButton() {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:26]: If method perform click, rename it like clickModalButton(). Also, fix description

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+        core().click();
+    }
+
+    @JDIAction("Is '{name}' opened")
+    public boolean isInnerModalOpened() {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:31]: Please remove method

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+        return core().hasClass(innerModalPage);
+    }
+
+    @JDIAction("Get '{name}'")
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:35]: fix description

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 22]
```diff
@@ -0,0 +1,62 @@
...
+    public boolean verifyModalTitle(String actualTitle) {
+        return core().find(modalTitle).getText().equals(actualTitle);
+    }
+
+    @JDIAction("Get '{name}'")
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:40]: fix description

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 23]
```diff
@@ -0,0 +1,62 @@
...
+    public boolean verifyModalDescription(String actualDescription) {
+        return core().find(modalDescription).getText().equals(actualDescription);
+    }
+
+    @JDIAction("Get '{name}'")
+    public void getInnerModalWindow() {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:46]: Please remove method

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Override
+    public void setup(Field field) {
+        if (!fieldHasAnnotation(field, JDIModal.class, Modal.class))
+            return;
+        JDIModal j = field.getAnnotation(JDIModal.class);
+        modalTitle = j.modalTitle();
+        modalDescription = j.modalDescription();
+        modalInnerButton = j.modalInnerButton();
+        innerModalPage = j.innerModalPage();
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:58]: Please remove 
        innerModalPage = j.innerModalPage();

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 25]
```diff
@@ -0,0 +1,62 @@
...
+import java.lang.reflect.Field;
+
+import static com.epam.jdi.light.elements.pageobjects.annotations.objects.FillFromAnnotationRules.fieldHasAnnotation;
+
+public class Modal extends UIBaseElement<ModalAssert> implements HasClick, ISetup {
+
+    String modalTitle;
+    String modalDescription;
+    String modalInnerButton;
+    String innerModalPage;
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:23]: Please remove innerModalPage and rename modalInnerButton to modalButton

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/annotations/JDIModal.java [hunk 26]
```diff
@@ -0,0 +1,19 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.FIELD})
+public @interface JDIModal {
+
+    @MarkupLocator String modalTitle() default "simple-modal-title";
+    @MarkupLocator String modalDescription() default "simple-modal-description";
+    @MarkupLocator String modalInnerButton() default "//button[contains('type=\"button\"')]";
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/annotations/JDIModal.java:16]: Please rename from modalInnerButton to modalButton

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/annotations/JDIModal.java [hunk 27]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.FIELD})
+public @interface JDIModal {
+
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/annotations/JDIModal.java:13]: Please add root() field

File: jdi-light-material-ui-tests/src/main/java/io/github/com/StaticSite.java [hunk 28]
```diff
@@ -201,6 +202,18 @@
...
+    public static ModalFrame modalFrame;
+
+    @Url("/material-ui-utils-modal--simple-modal")
+    public static WebPage simpleModalPage;
+
+    @Url("/material-ui-utils-modal--transitions-modal")
+    public static WebPage transitionsModalPage;
+
+    @Url("/story/material-ui-utils-modal--spring-modal")
+    public static WebPage springModalPage;
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/main/java/io/github/com/StaticSite.java:215]: Please remove unused WebPage

File: jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java [hunk 29]
```diff
@@ -0,0 +1,14 @@
...
+import com.epam.jdi.light.material.annotations.JDIModal;
+import com.epam.jdi.light.material.elements.utils.Modal;
+
+public class ModalFrame extends Section {
+
+    @JDIModal(modalRoot = "makeStyles-paper-1")
+    public Modal modal;
+
+    @JDIModal(buttonModalRoot = "#root > div > button")
+    public Modal buttonModal;
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java:13]: Please change type to Button

File: jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java [hunk 30]
```diff
@@ -0,0 +1,14 @@
+package io.github.com.pages.utils;
+
+import com.epam.jdi.light.elements.composite.Section;
+import com.epam.jdi.light.material.annotations.JDIModal;
+import com.epam.jdi.light.material.elements.utils.Modal;
+
+public class ModalFrame extends Section {
+
+    @JDIModal(modalRoot = "makeStyles-paper-1")
+    public Modal modal;
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java:10]: Please add different modal for different tests

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java [hunk 31]
```diff
@@ -0,0 +1,47 @@
...
+
+public class ModalTests extends TestsInit {
+
+    private static final String expectedSimpleTitle = "Text in a modal";
+    private static final String expectedSimpleDescription = "Duis mollis, est non commodo luctus, nisi erat porttitor ligula.";
+    private static final String expectedTransitionTitle = "Transition modal";
+    private static final String expectedTransitionDescription = "react-transition-group animates me.";
+    private static final String expectedServerTitle = "Server-side modal";
+    private static final String expectedServerDescription = "If you disable JavaScript, you will still see me.";
+
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/utils/ModalTests.java:19]: Please change names for constants

File: jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java [hunk 32]
```diff
@@ -0,0 +1,14 @@
+package io.github.com.pages.utils;
+
+import com.epam.jdi.light.elements.composite.Section;
+import com.epam.jdi.light.material.annotations.JDIModal;
+import com.epam.jdi.light.material.elements.utils.Modal;
+
+public class ModalFrame extends Section {
+
+    @JDIModal(modalRoot = "makeStyles-paper-1")
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java:9]: Please add all parameters to annotation

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java [hunk 33]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingServerTitle(String actualServerTitle) {
+        jdiAssert(element().verifyServerModalTitle(actualServerTitle), Matchers.is(true));
+        return this;
+    }
+
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingSimpleDescription(String actualSimpleDescription) {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java:31]: Please replace it with method verifyingModalDescription()

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java [hunk 34]
```diff
@@ -0,0 +1,47 @@
...
+        jdiAssert(element().verifySimpleModalDescription(actualSimpleDescription), Matchers.is(true));
+        return this;
+    }
+
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingTransitionDescription(String actualTransitionDescription) {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java:37]: Please replace it with method verifyingModalDescription()

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java [hunk 35]
```diff
@@ -0,0 +1,47 @@
...
+        jdiAssert(element().verifyTransitionModalDescription(actualTransitionDescription), Matchers.is(true));
+        return this;
+    }
+
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingServerDescription(String actualServerDescription) {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java:43]: Please replace it with method verifyingModalDescription()

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java [hunk 36]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingTransitionTitle(String actualTransitionTitle) {
+        jdiAssert(element().verifyTransitionModalTitle(actualTransitionTitle), Matchers.is(true));
+        return this;
+    }
+
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingServerTitle(String actualServerTitle) {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java:25]: Please replace it with method verifyingModalTitle()

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java [hunk 37]
```diff
@@ -0,0 +1,47 @@
...
+public class ModalAssert extends UIAssert<ModalAssert, Modal> {
+
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingSimpleTitle(String actualSimpleTitle) {
+        jdiAssert(element().verifySimpleModalTitle(actualSimpleTitle), Matchers.is(true));
+        return this;
+    }
+
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingTransitionTitle(String actualTransitionTitle) {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java:19]: Please replace it with method verifyingModalTitle()

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java [hunk 38]
```diff
@@ -0,0 +1,47 @@
...
+import com.epam.jdi.light.common.JDIAction;
+import com.epam.jdi.light.material.elements.utils.Modal;
+import org.hamcrest.Matchers;
+
+import static com.epam.jdi.light.asserts.core.SoftAssert.jdiAssert;
+
+public class ModalAssert extends UIAssert<ModalAssert, Modal> {
+
+    @JDIAction("Assert '{name}")
+    public ModalAssert verifyingSimpleTitle(String actualSimpleTitle) {
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/utils/ModalAssert.java:13]: Please replace it with method verifyingModalTitle()

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 39]
```diff
@@ -0,0 +1,101 @@
...
+
+public class Modal extends UIBaseElement<ModalAssert> implements HasClick, ISetup {
+
+    private static final String BUTTON = "#root > div > button";
+    private static final String SIMPLE_MODAL_TITLE = "#simple-modal-title";
+    private static final String SIMPLE_MODAL_DESCRIPTION = "#simple-modal-description";
+    private static final String TRANSITION_MODAL_TITLE = "#transition-modal-title";
+    private static final String TRANSITION_MODAL_DESCRIPTION = "#transition-modal-description";
+    private static final String SERVER_MODAL_TITLE = "#server-modal-title";
+    private static final String SERVER_MODAL_DESCRIPTION = "#server-modal-description";
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:22]: Please delete all these strings and use annotations fields in methods. Also, create two methods for verifying description and  title instead of getSimpleModalTitle(), getTransitionModalTitle(), getServerModalTitle(), getSimpleModalDescription(), getTransitionModalDescription(), getServerModalDescription(). The same thing for verifySimpleModalTitle(), verifyTransitionModalTitle(), verifyServerModalTitle() and etc

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 40]
```diff
@@ -0,0 +1,101 @@
...
+        return new ModalAssert().set(this);
+    }
+
+    @Override
+    public void setup(Field field) {
+        if (!fieldHasAnnotation(field, JDIModal.class, Modal.class))
+            return;
+        JDIModal j = field.getAnnotation(JDIModal.class);
+        modalTitle = j.modalTitle();
+        modalDescription = j.modalDescription();
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:97]: Please add modalRoot, buttonModalRoot

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/annotations/JDIModal.java [hunk 41]
```diff
@@ -0,0 +1,17 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.FIELD})
+public @interface JDIModal {
+    @MarkupLocator String modalRoot() default "";
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/annotations/JDIModal.java:13]: Please rename to root()

File: jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java [hunk 42]
```diff
@@ -0,0 +1,37 @@
...
+    public Button innerButton;
+
+    @JDIModal(additionalModal = "body > div:nth-child(10) > div.makeStyles-paper-1")
+    public Modal additionalModal;
+
+    @JDIModal(modalTitle = "#transition-modal-title")
+    public Modal transitionModalTitle;
+
+    @JDIModal(modalDescription = "#transition-modal-description")
+    public Modal transitionModalDescription;
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui-tests/src/main/java/io/github/com/pages/utils/ModalFrame.java:27]: Please join transitionModalDescription and transitionModalTitle to one element. The same for simpleModalTitle and simpleModalDescription, serverModalTitle and serverModalDescription

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java [hunk 43]
```diff
@@ -0,0 +1,61 @@
...
+import java.lang.reflect.Field;
+
+import static com.epam.jdi.light.elements.pageobjects.annotations.objects.FillFromAnnotationRules.fieldHasAnnotation;
+
+public class Modal extends UIBaseElement<ModalAssert> implements HasClick, ISetup {
+
+    String modalTitle;
+    String modalDescription;
+    String root;
+    String additionalModal;
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/utils/Modal.java:19]: Please remove additionalModal, this is not a part of Modal element

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/annotations/JDIModal.java [hunk 44]
```diff
@@ -0,0 +1,17 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.FIELD})
+public @interface JDIModal {
+    @MarkupLocator String root() default "";
+    @MarkupLocator String modalTitle() default "";
+    @MarkupLocator String modalDescription() default "";
+    @MarkupLocator String additionalModal() default "";
```
Comment (AlinaSelyutina) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/annotations/JDIModal.java:16]: Please remove additionalModal
